channels/plugins.py

```diff
@@ -73,11 +73,12 @@ def department_upserted(self, department, overwrite):
             channel = None
         elif department.school and (overwrite or not channel):
             channel, _ = Channel.objects.update_or_create(
+                search_filter=f"department={department.department_id}",
                 name=slugify(department.name),
```

Comment:
Remove this line or there will be channels for both the old and new name

Might be good to add a unit test for this too